src/osg/Texture.cpp

```diff
@@ -285,61 +315,102 @@ void Texture::TextureObject::setAllocated(GLint     numMipmapLevels,
+        case GL_COMPRESSED_RGB_S3TC_DXT1_EXT:              numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_SRGB_S3TC_DXT1_EXT:             numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_RGBA_S3TC_DXT1_EXT:             numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT1_EXT:       numBitsPerTexel = 4.0;  break;
+
+        case GL_COMPRESSED_RGB_ARB:                        numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_RGBA_S3TC_DXT3_EXT:             numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT3_EXT:       numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_RGBA_S3TC_DXT5_EXT:             numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_SRGB_ALPHA_S3TC_DXT5_EXT:       numBitsPerTexel = 8.0;  break;
+
+        case GL_COMPRESSED_SIGNED_RED_RGTC1_EXT:           numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_RED_RGTC1_EXT:                  numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_SIGNED_RED_GREEN_RGTC2_EXT:     numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_RED_GREEN_RGTC2_EXT:            numBitsPerTexel = 8.0;  break;
+
+        case GL_COMPRESSED_RGB_PVRTC_2BPPV1_IMG:           numBitsPerTexel = 2.0;  break;
+        case GL_COMPRESSED_RGBA_PVRTC_2BPPV1_IMG:          numBitsPerTexel = 2.0;  break;
+        case GL_COMPRESSED_RGB_PVRTC_4BPPV1_IMG:           numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_RGBA_PVRTC_4BPPV1_IMG:          numBitsPerTexel = 4.0;  break;
+
+        case GL_ETC1_RGB8_OES:                             numBitsPerTexel = 4.0;  break;
+
+        case GL_COMPRESSED_RGB8_ETC2:                      numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_SRGB8_ETC2:                     numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_RGB8_PUNCHTHROUGH_ALPHA1_ETC2:  numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_SRGB8_PUNCHTHROUGH_ALPHA1_ETC2: numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_RGBA8_ETC2_EAC:                 numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_SRGB8_ALPHA8_ETC2_EAC:          numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_R11_EAC:                        numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_SIGNED_R11_EAC:                 numBitsPerTexel = 4.0;  break;
+        case GL_COMPRESSED_RG11_EAC:                       numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_SIGNED_RG11_EAC:                numBitsPerTexel = 8.0;  break;
+
+        // ASTC
+        case GL_COMPRESSED_RGBA_ASTC_4x4_KHR:
+        case GL_COMPRESSED_SRGB8_ALPHA8_ASTC_4x4_KHR:      numBitsPerTexel = 8.0;  break;
+        case GL_COMPRESSED_RGBA_ASTC_5x4_KHR:
+        case GL_COMPRESSED_SRGB8_ALPHA8_ASTC_5x4_KHR:      numBitsPerTexel = 6.4;  break;
+        case GL_COMPRESSED_RGBA_ASTC_5x5_KHR:
+        case GL_COMPRESSED_SRGB8_ALPHA8_ASTC_5x5_KHR:      numBitsPerTexel = 5.12; break;
```

Comment:
This really doesn't feel like a good solution.  It's block compression, each block is always fits to a defined alignment of the block, it's not a ratio.

A better way would be to specify the block size and the dimensions of the xy block, then dived the widh, heigh by the number of pixels per block, round to the block size then compute the final size.

Reply:
I've tested this solution, It works for different astc images in KTX container.


Maybe just to add such comments:
```

case GL_COMPRESSED_SRGB8_ALPHA8_ASTC_5x4_KHR:      numBitsPerTexel = 6.4;  break;   // 128.0/(5*4) = 6.4
case GL_COMPRESSED_SRGB8_ALPHA8_ASTC_5x5_KHR:      numBitsPerTexel = 5.12; break;   // 128.0/(5*5) = 5.12

```
?

There is a table (**Block size -> Bits per pixel**) in specs:
https://www.khronos.org/registry/OpenGL/extensions/KHR/KHR_texture_compression_astc_hdr.txt
https://www.khronos.org/opengl/wiki/ASTC_Texture_Compression

Reply:
That doesn't fix my concerns, it's the wrong way to compute sizes required.  The x and y dimensions both have to be independently clamped to the block size.